### ISSUE TITLE
xgb_model_gemm does not give same result as xgb_model for Bosch

### ISSUE THREAD

**Issue by marsupialtail** [microsoft/hummingbird#531]
I tried to use hummingbird on the Bosch dataset, and I noticed that the xgb_model_gemm doesn't give the same results as real xgboost. Is this because there are nans in this dataset?

import pandas as pd
a = pd.read_csv("subbosch.csv")
X = a[[i for i in a.columns if "L" in i]].to_numpy()
y = a["Response"].to_numpy()
dtrain = xgb.DMatrix(X, label=y)
dtest = xgb.DMatrix(X, label=y)
xgb_model = xgb.XGBRegressor(nthread = 4, n_estimators=100, tree_method = "hist")
xgb_model.fit(X, y)
xgb_model_tree = convert(xgb_model, 'torch', X[0:1])
xgb_model_gemm = convert(xgb_model, 'torch', X[0:1],extra_config = {"tree_implementation":"gemm"})

assert xgb_model.predict(X[0:1]) == xgb_model_gemm.predict(X[0:1])

**Comment by interesaaat** [comment]
Yes you have to first do imputation if you want to use Hummingbird on Bosch. We actually have an implementation supporting missing values, but it is not currently in the main branch.

**Comment by marsupialtail** [comment]
Can you briefly comment on how you would do that? I suppose you can kind of just ignore the Nan value features in the first matrix?
